src/api/app/models/staging/staged_requests.rb

```diff
@@ -91,6 +91,9 @@ def link_package(bs_request_action)
     source_package = Package.get_by_project_and_name!(bs_request_action.source_project,
                                                       bs_request_action.source_package)
 
+    target_package = Package.get_by_project_and_name!(bs_request_action.target_project,
```

Comment:
Make sure you spec this for 'if the target package does not yet exist'

Reply:
Actually on second thought: the target project has no business here at all. All spec files are required to be linked - if the submission introduces new ones, those as well.

Reply:
@coolo `target_project` or `target_package` ?

Reply:
`target_project` - the target_package is the name, but within the staging project.

Reply:
Let's see, if my rational is right:

I'm using the `target_project` + `target_package` here, because in the `curl` package on `openSUSE:Factory` (which will normally will be our `target_project` ) is where I'm going to find the local links, i.e for curl will be the `curl-mini`

Reply:
the point is *it will be* - if you submit curl-ng to openSUSE:Factory, the staging project still needs to linkpac curl-ng-mini to curl-ng

Same as submitting curl-now-as-mutlibuild should *not* create a curl-mini in the staging project. The target project's local links have *zero* influence on the local links in staging projects.

Reply:
I couldn't really follow your example, but The last sentence was key to me. 

"The target project's local links have zero influence on the local links in staging projects." is the opposite that we were assuming. Could you please describe me (maybe as issue, if a comment isn't the best place for it) how exactly those links should be built? Today I'm just able to reverse engineer it https://build.opensuse.org/package/show/openSUSE:Factory/curl-mini

Reply:
@coolo well, creating new local links (in staging or target) based on the existence of build descriptions in the source is not a route I want to go down with this. This is too distro specific for my taste. That you would have to do on your own with some bot or something, sorry.

Reply:
I don't believe that local links are more distribution specific than staging projects in general. But that also means I can't use the accept command of a staging project, but need to do everything from the client. That's a pity after all the work put into accepting.

Reply:
@coolo let's discuss this offline please

Reply:
sure

Reply:
I actually wonder when "let's discuss this offline please" became "I leave for vaction, let's merge this anyway".

Reply:
Probably, he assumed that because it's not in production, we could iterate through it again until we get it right, and it shouldn't block our deliverable (because a specification change/misunderstanding in the day that we actually should finish the deliverable) 

If you could take part in the planning meetings, or find a way to spec it in a proper way, maybe we could avoid those issues..  PR based development, where you specify the feature when the PR is ready, probably won't work.

Reply:
let's discuss this offline please